library/src/main/java/com/pokegoapi/main/RequestHandler.java

```diff
@@ -274,68 +198,67 @@ private Long getRequestId() {
 
 	@Override
 	public void run() {
-		List<AsyncServerRequest> requests = new LinkedList<>();
+		AsyncServerRequest<GeneratedMessage, Object> request = null;
 		AuthTicket authTicket = null;
 		while (true) {
 			try {
-				Thread.sleep(350);
-			} catch (InterruptedException e) {
-				throw new AsyncPokemonGoException("System shutdown", e);
+				request = workQueue.take();
+			} catch (Throwable ignored) {
+				// Ignore
 			}
-			if (workQueue.isEmpty()) {
+
+			if (request == null) {
 				continue;
 			}
 
-			workQueue.drainTo(requests);
+			ArrayList<InternalServerRequest> serverRequests = new ArrayList<>();
```

Comment:
Why ArrayList on the left hand, why no size given to the ArrayList on the right hand, since the size is known